openapi-core/src/main/java/io/ballerina/openapi/core/generators/client/FunctionSignatureGenerator.java

```diff
@@ -574,10 +591,32 @@ private String getRequestBodyParameterForArraySchema(String operationId, Map.Ent
             // TODO - Add API doc by checking requestBody
             TypeDefinitionNode arrayTypeNode =
                     ballerinaSchemaGenerator.getTypeDefinitionNode(arraySchema, paramType, new ArrayList<>());
-            functionReturnType.updateTypeDefinitionNodeList(paramType, arrayTypeNode);
+            updateTypeDefinitionNodeList(paramType, arrayTypeNode);
         } else {
             paramType = GeneratorUtils.getBallerinaMediaType(next.getKey().trim()) + SQUARE_BRACKETS;
         }
         return paramType;
     }
+
+    /**
+     * This util function for update the typeDefinition node after check it duplicates.
+     *
+     * @param typeName    - Given Node name
+     * @param typeDefNode - Generated Node
+     */
+    private void updateTypeDefinitionNodeList(String typeName, TypeDefinitionNode typeDefNode) {
+        boolean isExit = false;
+        if (!typeDefinitionNodeList.isEmpty()) {
+            for (TypeDefinitionNode typeNode : typeDefinitionNodeList) {
+                if (typeNode.typeName().toString().trim().equals(typeName)) {
+                    isExit = true;
+                }
+            }
+            if (!isExit) {
+                typeDefinitionNodeList.add(typeDefNode);
+            }
+        } else {
+            typeDefinitionNodeList.add(typeDefNode);
```

Comment:
In here we can use the streams APIs (`anyMatch()` API) on the typeDefinitionNodeList to simplify the logic

Reply:
Updated. Please review